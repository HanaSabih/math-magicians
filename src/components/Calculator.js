import React, { useState } from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import { Container, Stack } from 'react-bootstrap';

import calculate from '../logic/calculate';

const Calculator = () => {
  const [state, setState] = useState({
    total: '0',
    operation: null,
    next: null,
  });

  const clickBtn = (btn) => {
    setState((state) => calculate(state, btn.target.name));
  };

  const { total, next, operation } = state;

  return (

    <Container className="w-25 mt-5  ">
      <Stack direction="horizontal" className="mx-auto bg-secondary resultContainer">
        <div className="text-end w-100 bg-secondary text-white px-3">
          <span>{total}</span>
          <span>{operation}</span>
          <span>{next}</span>
        </div>
      </Stack>
      <Stack direction="horizontal" className="mx-auto">
        <button type="button" name="AC" onClick={clickBtn}>AC</button>
        <button type="button" name="+/-" onClick={clickBtn}>+/-</button>
        <button type="button" name="%" onClick={clickBtn}>%</button>
        <button type="button" name="รท" className="orangeBtn" onClick={clickBtn}>รท</button>
      </Stack>
      <Stack direction="horizontal" className="mx-auto">
        <button type="button" name="7" onClick={clickBtn}>7</button>
        <button type="button" name="8" onClick={clickBtn}>8</button>
        <button type="button" name="9" onClick={clickBtn}>9</button>
        <button type="button" name="x" className="orangeBtn" onClick={clickBtn}>x</button>
      </Stack>
      <Stack direction="horizontal" className="mx-auto">
        <button type="button" name="4" onClick={clickBtn}>4</button>
        <button type="button" name="5" onClick={clickBtn}>5</button>
        <button type="button" name="6" onClick={clickBtn}>6</button>
        <button type="button" name="-" className="orangeBtn" onClick={clickBtn}>-</button>
      </Stack>
      <Stack direction="horizontal" className="mx-auto">
        <button type="button" name="1" onClick={clickBtn}>1</button>
        <button type="button" name="2" onClick={clickBtn}>2</button>
        <button type="button" name="3" onClick={clickBtn}>3</button>
        <button type="button" name="+" className="orangeBtn" onClick={clickBtn}>+</button>
      </Stack>
      <Stack direction="horizontal" className="mx-auto">
        <button type="button" name="0" className="col-md-6" onClick={clickBtn}>0</button>
        <button type="button" name="." onClick={clickBtn}>.</button>
        <button type="button" name="=" className="orangeBtn" onClick={clickBtn}>=</button>
      </Stack>
    </Container>
  );
};
export default Calculator;
