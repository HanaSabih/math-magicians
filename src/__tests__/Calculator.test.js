import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

it('renders correctly ', () => {
  const testQuote = renderer.create(<Calculator />).toJSON();
  expect(testQuote).toMatchSnapshot();
});
