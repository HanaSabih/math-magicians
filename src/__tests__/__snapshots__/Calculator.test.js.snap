// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly  1`] = `
<div
  className=" mx-auto homeCalc container"
>
  <div
    className="row"
  >
    <div
      className="col-md-4 col-12"
    >
      <div
        className=" mt-5 "
      >
        <h1>
          Let's do some Math!
        </h1>
      </div>
    </div>
    <div
      className="col-md-8 col-12"
    >
      <div
        className=" mt-5  pb-3  container"
      >
        <div
          className="mx-auto bg-secondary resultContainer fs-1 hstack"
        >
          <div
            className="text-end w-100 bg-secondary text-white px-3"
          >
            <span>
              0
            </span>
            <span />
            <span />
          </div>
        </div>
        <div
          className="mx-auto fs-3 hstack"
        >
          <button
            name="AC"
            onClick={[Function]}
            type="button"
          >
            AC
          </button>
          <button
            name="+/-"
            onClick={[Function]}
            type="button"
          >
            +/-
          </button>
          <button
            name="%"
            onClick={[Function]}
            type="button"
          >
            %
          </button>
          <button
            className="orangeBtn"
            name="รท"
            onClick={[Function]}
            type="button"
          >
            รท
          </button>
        </div>
        <div
          className="mx-auto fs-3 hstack"
        >
          <button
            name="7"
            onClick={[Function]}
            type="button"
          >
            7
          </button>
          <button
            name="8"
            onClick={[Function]}
            type="button"
          >
            8
          </button>
          <button
            name="9"
            onClick={[Function]}
            type="button"
          >
            9
          </button>
          <button
            className="orangeBtn"
            name="x"
            onClick={[Function]}
            type="button"
          >
            x
          </button>
        </div>
        <div
          className="mx-auto fs-3 hstack"
        >
          <button
            name="4"
            onClick={[Function]}
            type="button"
          >
            4
          </button>
          <button
            name="5"
            onClick={[Function]}
            type="button"
          >
            5
          </button>
          <button
            name="6"
            onClick={[Function]}
            type="button"
          >
            6
          </button>
          <button
            className="orangeBtn"
            name="-"
            onClick={[Function]}
            type="button"
          >
            -
          </button>
        </div>
        <div
          className="mx-auto fs-3 hstack"
        >
          <button
            name="1"
            onClick={[Function]}
            type="button"
          >
            1
          </button>
          <button
            name="2"
            onClick={[Function]}
            type="button"
          >
            2
          </button>
          <button
            name="3"
            onClick={[Function]}
            type="button"
          >
            3
          </button>
          <button
            className="orangeBtn"
            name="+"
            onClick={[Function]}
            type="button"
          >
            +
          </button>
        </div>
        <div
          className="mx-auto fs-3 hstack"
        >
          <button
            className="col-6 "
            name="0"
            onClick={[Function]}
            type="button"
          >
            0
          </button>
          <button
            name="."
            onClick={[Function]}
            type="button"
          >
            .
          </button>
          <button
            className="orangeBtn"
            name="="
            onClick={[Function]}
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
